include/diy/master.hpp

```diff
@@ -1068,23 +1099,20 @@ flush()
     if (cur - start > wait*1000)
     {
         log->notice("Waiting in flush [{}]: {} - {} out of {}",
-                    comm_.rank(), inflight_sends_.size(), received_, expected_);
+                    comm_.rank(), inflight_sends_.size(), incoming_[exchange_round_].received, expected_);
         wait *= 2;
     }
 #endif
-  } while (!inflight_sends_.empty() || received_ < expected_ || !to_send.empty());
+  } while (!inflight_sends_.empty() || incoming_[exchange_round_].received < expected_ || !to_send.empty());
 
   outgoing_.clear();
 
   log->debug("Done in flush");
   //show_incoming_records();
 
   process_collectives();
-  prof << "barrier";
-  comm_.barrier();
-  prof >> "barrier";
 
-  received_ = 0;
+  incoming_[exchange_round_].received = 0;
```

Comment:
We probably don't want this. Before it was resetting `received_` for the next round. It's probably a good idea to keep it unchanged for debugging purposes.